boot/boot_serial/src/boot_serial.c

```diff
@@ -446,6 +447,7 @@ bs_echo(char *buf, int len)
     cbor_encode_text_string(&bs_rsp, echo_buf, echo_buflen);
     cbor_encoder_close_container(&bs_root, &bs_rsp);
     boot_serial_output();
+    bs_echoreceived = 1;
```

Comment:
maybe you can use for that any command we already have, for instance the `image list` ( bs_list()) command. I think no obstacles to do that.